components/activecampaign/sources/contact-added-to-list/contact-added-to-list.js

```diff
@@ -6,9 +6,57 @@ module.exports = {
   name: "New Contact Added to List",
   key: "activecampaign-contact-added-to-list",
   description: "Emits an event each time a new contact is added to a list.",
-  version: "0.0.1",
+  version: "0.0.2",
+  props: {
+    ...common.props,
+    lists: { propDefinition: [activecampaign, "lists"] },
+  },
+  hooks: {
+    async activate() {
+      const sources =
+        this.sources.length > 0
+          ? this.sources
+          : this.activecampaign.getAllSources();
+      const hookIds = [];
+      const events = this.getEvents();
+      if (this.lists.length > 0) {
+        for (const list of this.lists) {
+          const { id } = await this.getHookData(
+            events,
+            this.http.endpoint,
+            sources,
+            list
+          );
+          hookIds.push(id);
+        }
```

Comment:
In case we're unable to create a new webhook, we should delete the webhooks that we created so far.
Otherwise the event source will leave the user's account in an inconsistent state.